pandas/tests/tools/test_to_datetime.py

```diff
@@ -71,6 +71,20 @@ def test_to_datetime_format(self, cache):
                 else:
                     tm.assert_index_equal(result, expected)
 
+    def test_to_datetime_format_some_invalid_numbers(self):
```

Comment:
you added a ton of conditionals, i would expect that you can parameterize so that each is exercised (or simply not needed)